apis/keda/v1alpha1/scaledobject_types.go

```diff
@@ -167,3 +169,19 @@ func init() {
 func (so *ScaledObject) GenerateIdentifier() string {
 	return GenerateIdentifier("ScaledObject", so.Namespace, so.Name)
 }
+
+func (so *ScaledObject) HasPausedAnnotation() bool {
+	_, pausedAnnotationFound := so.GetAnnotations()[PausedAnnotation]
+	_, pausedReplicasAnnotationFound := so.GetAnnotations()[PausedReplicasAnnotation]
+	return pausedAnnotationFound || pausedReplicasAnnotationFound
+}
+
+func (so *ScaledObject) NeedPaused() bool {
```

Comment:
Could you please rename this method to be more clear what is it about? also could you please add a comment with short description of the functionality?